app/src/main/java/com/sopt/dive/presentation/signin/SignInScreen.kt

```diff
@@ -0,0 +1,222 @@
+package com.sopt.dive.presentation.signin
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.sopt.dive.R
+import com.sopt.dive.core.designsystem.component.DiveButton
+import com.sopt.dive.core.designsystem.component.LabelTextField
+import com.sopt.dive.core.designsystem.component.PasswordTextField
+import com.sopt.dive.core.designsystem.theme.DiveTheme
+import com.sopt.dive.core.local.datastore.UserDataStore
+import com.sopt.dive.core.util.noRippleClickable
+import com.sopt.dive.core.util.showToast
+import kotlinx.coroutines.launch
+
+@Composable
+fun SignInRoute(
+    resultUserState: Pair<String, String>?,
+    userDataStore: UserDataStore,
+    navigateToSignUp: () -> Unit,
+    navigateToMain: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    var userId by rememberSaveable { mutableStateOf("") }
+    var password by rememberSaveable { mutableStateOf("") }
+    var isLoading by remember { mutableStateOf(true) }
+
+    val coroutineScope = rememberCoroutineScope()
+    val context = LocalContext.current
+
+    LaunchedEffect(resultUserState) {
+        if (resultUserState != null) {
+            userId = resultUserState.first
+            password = resultUserState.second
+        }
+    }
```

Comment:
사소하지만 early return을 쓰는것도 가독성에 좋을 것 같네요 :)
```suggestion
    LaunchedEffect(resultUserState) {
        if (resultUserState == null) return
        
        userId = resultUserState.first
        password = resultUserState.second
    }
```